modules/core/src/main/scala/org/scalasteward/core/scalafmt/ScalafmtAlg.scala

```diff
@@ -43,5 +49,21 @@ object ScalafmtAlg {
           scalafmtConfFile = repoDir / ".scalafmt.conf"
           fileContent <- fileAlg.readFile(scalafmtConfFile)
         } yield fileContent.flatMap(parseScalafmtConf)
+
+      override def runScalafmt(repo: Repo, mainArtifactId: String): F[Unit] =
+        for {
+          repoDir <- workspaceAlg.repoDir(repo)
+          shouldRunScalafmt <-
+            if (mainArtifactId === scalafmtArtifactId) {
+              repoConfigAlg
+                .readRepoConfigWithDefault(repo)
+                .map(_.scalafmt.exists(_.runAfterUpgrading))
+            } else {
+              F.pure(false)
+            }
+          _ <- F.whenA(shouldRunScalafmt) {
+            processAlg.execSandboxed(Nel.of("sbt", ";scalafmtAll;scalafmtSbt"), repoDir)
```

Comment:
Have you thought about using the [CLI version of Scalafmt](https://scalameta.org/scalafmt/docs/installation.html#cli) here. I think it could be as easy as `processAlg.exec(Nel.of("scalafmt"), repoDir)`. I see two advantages of using the CLI. It is probably faster than using sbt and it would also work with other build tools than sbt.

Reply:
Updated code to use `scalafmt` https://github.com/scala-steward-org/scala-steward/pull/1673/commits/061d2f9455878b23df2e622a0ca34ec597e8fd18